openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -0,0 +1,575 @@
+        C = self.COEFFS[imt]
+        C_PGA = self.COEFFS[PGA()]
+
+        m_b = self._get_mb()
+        c0, c0_pga = self._c0(C, C_PGA)
+        fm, fm_pga = self._magnitude_scaling(C, C_PGA, rup.mag, m_b)
+        fp, fp_pga = self._path_term(C, C_PGA, rup.mag, dists.rrup, m_b)
+        if hasattr(rup, 'hypo_depth'):
+            fd = self._depth_scaling(C, rup.hypo_depth)
+            fd_pga = self._depth_scaling(C_PGA, rup.hypo_depth)
+        else:
+            fd = 0
+            fd_pga = 0
+        fb = self._basin_term(C, sites)
+        flin = self._linear_amplification(C, sites.vs30)
+        fnl = self._non_linear_term(C, imt, sites.vs30, fp_pga, fm_pga, c0_pga,
+                                    fd_pga)
+
+        # The output is the desired median model prediction in LN units
+        # Take the exponential to get PGA, PSA in g or the PGV in cm/s
+        mean = fp + fnl + fb + flin + fm + c0 + fd
+
+        stddevs = self.get_stddevs(C, dists.rrup, sites.vs30, stddev_types)
+        return mean, stddevs
+
+    def get_stddevs(self, C, rrup, vs30, stddev_types):
+        """
+        Returns the standard deviations.
+        Generate tau, phi, and total sigma computed from both
+        total and partitioned phi models.
+        """
+
+        # define period-independent coefficients for phi models
+        v1 = 200
+        v2 = 500
+        r1 = 200
+        r2 = 500
+
+        stddevs = []
+
+        # total Phi
+        phi_rv = np.zeros(len(vs30))
+        for i, vs30i in enumerate(vs30):
+            if rrup[i] <= r1:
+                phi_rv[i] = C["phi21"]
+            elif rrup[i] >= r2:
+                phi_rv[i] = C["phi22"]
+            else:
+                phi_rv[i] = ((C["phi22"] - C["phi21"])
+                             / (math.log(r2) - math.log(r1))) \
+                            * (math.log(rrup[i]) - math.log(r1)) + C["phi21"]
+
+            if vs30i <= v1:
+                phi_rv[i] += C["phi2V"] * \
+                             (math.log(r2 / max(r1, min(r2, rrup[i])))
+                              / math.log(r2 / r1))
+            elif vs30i < v2:
+                phi_rv[i] += C["phi2V"] * \
+                          ((math.log(v2 / min(v2, vs30i)))
+                           / math.log(v2 / v1)) * \
+                          (math.log(r2 / max(r1, min(r2, rrup[i])))
+                           / math.log(r2 / r1))
+
+        phi_tot = np.sqrt(phi_rv)
+
+        for stddev in stddev_types:
+            assert stddev in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            if stddev == const.StdDev.TOTAL:
+                stddevs.append(np.sqrt(C["Tau"] ** 2 + phi_tot ** 2))
+            elif stddev == const.StdDev.INTER_EVENT:
+                stddevs.append(np.repeat(C["Tau"], len(vs30)))
+            elif stddev == const.StdDev.INTRA_EVENT:
+                stddevs.append(phi_tot)
+
+        return stddevs
+
+    def hypocentral_depth_estimate(self, ztor, width, dip):
```

Comment:
No, this is an alternative described in the paper if hypocentral depth is unknown. I left it in. Perhaps it could be a function at the top level or I can remove it entirely.

Reply:
I've gone ahead and removed it.